lib/xandra/connection.ex

```diff
@@ -341,13 +349,9 @@ defmodule Xandra.Connection do
       send(data.cluster_pid, {:xandra, :disconnected, data.peername, self()})
     end
 
-    data =
-      Enum.reduce(data.in_flight_requests, data, fn {stream_id, req_alias}, data_acc ->
-        send_reply(req_alias, {:error, :disconnected})
-        update_in(data_acc.free_stream_ids, &MapSet.put(&1, stream_id))
-      end)
-
-    data = put_in(data.in_flight_requests, %{})
+    Enum.each(data.in_flight_requests, fn {_stream_id, req_alias} ->
+      send_reply(req_alias, {:error, :disconnected})
+    end)
```

Comment:
I’m confused: now, we send the reply to the caller but we don't update the `in_flight_requests`, which we were resetting to `%{}` before. How does this work?

Reply:
True, I must've overseen this, I deleted it in another iteration.